src/xhci_backend.rs

```
@@ -124,6 +137,24 @@ impl ServerBackend for XhciBackend {
                 },
             ),
 
+            0 => self.device.write_io(
+                0,
+                RequestKind::Memory,
+                Request::new(offset, RequestSize::try_from(data.len() as u64).unwrap()),
```

For integer conversions I prefer `try_into().unwrap()`, because it will do the right thing or fail loudly. Especially for `usize -> u64` this has no cost for all platforms we care for.

Otherwise, for `unwrap()` I would appreciate a comment why `unwrap` is safe.

---

Yes, it does look much worse. :(